pkg/services/apiserver/aggregator/aggregator.go

```diff
@@ -198,16 +196,20 @@ func makeAPIServiceAvailableHealthCheck(name string, apiServices []*v1.APIServic
 	}
 
 	// Watch add/update events for APIServices
-	_, _ = apiServiceInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+	_, err := apiServiceInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
 		AddFunc:    func(obj interface{}) { handleAPIServiceChange(obj.(*v1.APIService)) },
 		UpdateFunc: func(old, new interface{}) { handleAPIServiceChange(new.(*v1.APIService)) },
 	})
```

Comment:
@toddtreece what happens if an APIService is added, never becomes ready and then gets removed?

IIUC this means that the healthcheck will never succeed because we will not remove the that service from the list of pending APIServices. If that's correct, we may want to have a delete handler here as well.

Reply:
@radiohead good question. this was copied from [kube-apiserver](https://github.com/kubernetes/kubernetes/blob/73f19e4c0162a2ee711c254b2af226b4e135d0e3/cmd/kube-apiserver/app/aggregator.go#L233), and the list of pending services used for the health check is only for the API groups that are registered on startup, which for `kube-apiserver` are groups like the core v1 API groups. services registered on startup are all local delegation, and bypass any ready checks and are considered always available. i think unless something is very broken right now it's probably not going to be an issue because the API group should be removed from the pending list during the startup process. @charandas this is a great observation, and something we should keep in mind if we end up needing to inject remote services into the list of startup groups.